### PR TITLE
Dance AI: adding hook name to missing hook warning

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
We have seen a large number of "Missing required hook in compiled code" warnings in Dance AI. This adds the hook name to the warning payload so we can narrow down which hooks are missing.

Warnings that we are currently seeing on Cloudwatch in the last week:
Screenshot 2023-12-01 at 9 06 48 AM

## Warning!!

We have entered Pixel Lock for Hour of Code!

Computer Science Education Week will be happening from Dec 4 - Dec 10. Alongside this event, we will
be launching our new Hour of Code activity. Please consider any risk introduced in this PR that
could affect Dance Lab, instructions, saving and logging student progress, caching, or anything
related to the Hour of Code activities, old or new. Even small changes, such as a different button
color, are considered significant during this time. If this change will affect the new Hour of Code
activity in any way, join the morning change review to get your changes approved prior to merging.
Reach out to the Student Labs team for more details!

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by bencodeorg** [comment]
I recall seeing this warning a lot in my dev console recently, are you not able to repro?

File: apps/src/dance/lab2/ProgramExecutor.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default class ProgramExecutor {
 
     this.hooks = await this.compileAllCode(code);
     if (!this.hooks.runUserSetup || !this.hooks.getCueList) {
-      Lab2MetricsReporter.logWarning('Missing required hooks in compiled code');
+      this.reportMissingHooks('runUserSetup', 'getCueList');
       return;
     }
```
Comment (breville) [inline apps/src/dance/lab2/ProgramExecutor.ts:84]: Worth splitting into two separate checks?

**Review by breville — APPROVED** [review]

**Comment by sanchitmalhotra126** [comment]
> I recall seeing this warning a lot in my dev console recently, are you not able to repro?

@bencodeorg oh interesting, I have not been able to repro - do you have a consistent repro locally?

**Comment by bencodeorg** [comment]
I of course don't see it now when playing around with the modal :). I don't recall when I saw it, but I'll keep an eye out!

File: apps/src/dance/lab2/ProgramExecutor.ts (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/src/dance/lab2/ProgramExecutor.ts:84]: Hmm we could, but both of these are necessary in conjunction to start code execution

Comment (breville) [reply apps/src/dance/lab2/ProgramExecutor.ts:84]: Oh yes, we'd have to return in either case.  It would just help differentiate in the logging.

**Review by jmkulwik — APPROVED** [review]

**Review by cat5inthecradle — APPROVED** [review]

**Pull request merged**